direct/core/src/test/java/cz/o2/proxima/direct/storage/InMemStorage.java

```diff
@@ -436,13 +455,15 @@ private void handleFlushDataBaseOnPosition(
             try {
               if (!killSwitch.get()) {
                 el = cloneAndUpdateAttribute(getEntityDescriptor(), el);
-                long off = offsetTracker.incrementAndGet();
                 watermark.update(el);
                 long w = watermark.getWatermark();
+                consumedOffsets.add(
+                    String.format("%s#%s:%d", el.getKey(), el.getAttribute(), el.getStamp()));
```

Comment:
String format is expensive for hot code paths, can we replace this with simple string concat?

Reply:
This is testing storage, performance is of no convern here.